### DISCUSSION TITLE
Does Categraf v0.2.29 have a Kafka version requirement?

### DISCUSSION THREAD

**Discussion by wushengwusheng** [ccfos/nightingale#1370] [Q&A] [answered]
categraf：v0.2.29
kafka：2.10-0.10.1.1
报错：failed to init input: local.kafka error: could not instantiate kafka lag exporter: kafka: client has run out of available brokers to talk to: EOF

请问这报错是不是最新版本对kafka版本有要求导致的？

**Comment by wushengwusheng** [comment]
![image](https://user-images.githubusercontent.com/30250875/217142724-7d4e31be-52bf-4d08-b481-6ca38c6e4ebe.png)

**Comment by kongfei605** [comment, marked as answer]
配置中有个选项kafka_version， 可以指定kafka的版本，配置注释有。

**Reply by wushengwusheng** [reply]
嗯看到了谢谢～～一开始报错，后来我又重新加了一个kafka集群就好了,但是我看categraf日志里一直在报main.go:106: I! received signal: broken pipe 这个日志不确定有没有影响或者是因为什么报的，大佬有空可以看看

**Reply by kongfei605** [reply]
容器环境还是systemd托管的？

**Reply by kongfei605** [reply]
sigpipe信号一般是写入关闭的socket，可以kill -HUP $PID 或者 重启categraf

**Reply by wushengwusheng** [reply]
systemd托管的
我发现一个现象
单单2.10.0版本的会报错failed to init input: local.kafka error: could not instantiate kafka lag exporter: kafka: client has run out of available brokers to talk to: EOF，然后在同一台机器上再加一个2.11.0版本的就不报错了可以正常汇报了，然后再把2.11版本的去掉重启categraf也是正常。
这个现象有点想不通

**Comment by wushengwusheng** [comment]
![image](https://user-images.githubusercontent.com/30250875/217446427-ae7f47c4-dfc9-46a9-86e4-239ba09b60f4.png)

**Reply by wushengwusheng** [reply]
![image](https://user-images.githubusercontent.com/30250875/217446569-7eaa207e-151f-4e7d-8e25-c1078b8066e8.png)

**Comment by wushengwusheng** [comment]
已解决，需要制定kafka_version，感谢孔飞大佬指导

**Comment by carry2017123** [comment]
可以解决，谢谢大佬
